pipelines/nir_equivalence.py

```
@@ -0,0 +1,1987 @@
+    elif len(executed) == 1:
+        execution = f"only one in-repo runtime executed: {executed[0]!r}"
+    else:
+        execution = "no in-repo runtime executed this graph"
+    return (
+        f"{execution}; nir-rs and the other upstream NIR runtimes were unavailable, "
+        "so this record is evidence only about the runtimes and diagnostics that "
+        "actually executed"
+    )
+
+
+def _evidence_digests(entries):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `_evidence_digests`

(ANN202)